profiling-ffi/src/profiles.rs

```diff
@@ -371,19 +372,47 @@ pub unsafe extern "C" fn ddog_prof_Profile_set_endpoint<'a>(
     profile.add_endpoint(local_root_span_id, endpoint);
 }
 
+/// Count the number of times an endpoint has been seen.
+///
+/// # Arguments
+/// * `profile` - a reference to the profile that will contain the samples.
+/// * `endpoint` - the endpoint label for which the count will be incremented
+///
+/// # Safety
+/// The `profile` ptr must point to a valid Profile object created by this
+/// module.
+/// This call is _NOT_ thread-safe.
+#[no_mangle]
+pub unsafe extern "C" fn ddog_prof_Profile_add_endpoint_count(
+    profile: &mut datadog_profiling::profile::Profile,
+    endpoint: CharSlice,
+    value: i64,
+) {
+    let endpoint = endpoint.to_utf8_lossy();
+    profile.add_endpoint_count(endpoint, value);
+}
+
 #[repr(C)]
 pub struct EncodedProfile {
     start: Timespec,
     end: Timespec,
     buffer: ddcommon_ffi::Vec<u8>,
+    endpoints_stats: Box<profiled_endpoints::ProfiledEndpointsStats>,
```

Comment:
Is the Box here needed to keep profiled_endpoints as an opaque pointer because it's not repr(C)-able?

Reply:
That and also release the profiled_endpoints when the encode profile is release